index.js

```diff
@@ -34,6 +33,29 @@ function load() {
   return result;
 }
 
+function extend(a, b) {
+  // iterate over all direct and inherited enumerable properties
+  for (const name in b) {
+    let newValue = b[name];
+
+    // check if the new value is an object and its property exists on the former
```

Comment:
This merging logic is more lenient than BCD needs, and would allow for weird errors in the BCD data which would go unnoticed unless there's a lint for it. This might be a good opportunity to simplify this to make more assumptions about the data, throwing exceptions if they're not met.

I think that any time a property exists on both `a` and `b`, both values must be regular objects (not arrays, for example), otherwise something is wrong.

Reply:
Doing this will reveal that there is some broken structure in BCD, the first one I run into is "bcd.css.selectors.not_match_link" defined in multiple files by accident.

Reply:
I've fixed that in https://github.com/mdn/browser-compat-data/pull/9842.

Reply:
> This might be a good opportunity to simplify this to make more assumptions about the data, throwing exceptions if they're not met.

Yes. I’d like to help here, if I can. 😄

This PR aims to provide a simplified method that would not throw new errors, and then also honor the ** peculiar** bits from the (jQuery) extend dependency implementation; versus something like a deep `Object.assign`.

I see two options, and defer to your preference. One option is to "_do the most_", and improve the usefulness of this function and then bundle any resulting fixes that are surfaced. The other option is to "_do the least_", and approve or improve this function to replace the existing dependency, with its **peculiar** bits and leniencies out in the open.

Reply:
With that fixed, here's a less forgiving variant:

```js
function isPlainObject(v) {
  return typeof v === 'object' && v !== null && !Array.isArray(v);
}

function extend(target, source) {
  if (!isPlainObject(target) || !isPlainObject(source)) {
    throw new Error('Both target and source must be plain objects');
  }
  // iterate over all enumerable properties
  for (const [key, value] of Object.entries(source)) {
    // recursively extend if target has the same key, otherwise just assign
    if (Object.prototype.hasOwnProperty.call(target, key)) {
      extend(target[key], value);
    } else {
      target[key] = value;
    }
  }

  return target;
}
```

Feel free to use as much or as little of that as you like.

Reply:
Sorry, I commented before I read your comment. A deep `Object.assign` is indeed what I think we should use here, something that throws if used with arrays in particular. `Object.assign({}, null)` doesn't throw, but that's a case we should also disallow to catch more errors I think.

Reply:
Hi, @foolip! After rebasing with the latest `main` branch and updating the `package-lock.json` file, everything continues to run as expected.

However, if I update the `extend` function with the _less forgiving variant_ then the `lint` command fails.

The error is rather catastrophic on a terminal as the linter attempts to whole compat objects as being invalid, but it starts like this:

```
✖ api/ANGLE_instanced_arrays.json
  JSON Schema – 1 error:
```
  
I suspect the deep mutation of `target` causes this error, and _something else_ expects deep objects not to change.

I also tried updating the function to _always_ return a new plain object, and the lint command failed again. This time, I suspect the missing shallow mutation of `target` causes this error! Haha. 

This place in the middle works, and is done in https://github.com/mdn/browser-compat-data/pull/9821/commits/bde56ca3b3b8040b17f4f9c8a1d97a96c6ff008b 🎉

```js
function isPlainObject(v) {
  return typeof v === 'object' && v !== null && !Array.isArray(v);
}

function extend(target, source) {
  if (!isPlainObject(target) || !isPlainObject(source)) {
    throw new Error('Both target and source must be plain objects');
  }

  // iterate over all enumerable properties
  for (const [key, value] of Object.entries(source)) {
    // recursively extend if target has the same key, otherwise just assign
    if (Object.prototype.hasOwnProperty.call(target, key)) {
      target[key] = extend(extend({}, target[key]), value);
    } else {
      target[key] = value;
    }
  }

  return target;
}
```

Of course, I’m glad to continue working with you on this, if you’d like to remove this last **peculiar** bit. 😄